examples/material_gallery/lib/demo/grid_list_demo.dart

```diff
@@ -54,21 +54,38 @@ class GridDemoPhotoItem extends StatelessWidget {
     Set<Key> mostValuableKeys = new HashSet<Key>();
     mostValuableKeys.add(photoKey);
 
-    Navigator.push(context, new MaterialPageRoute<Null>(
+    RenderBox photoBox = context.findRenderObject();
+    RenderBox navigatorBox = ModalRoute.of(context).navigator.context.findRenderObject();
+    Size navigatorSize = navigatorBox.size;
+    Point pixelCenter = navigatorBox.globalToLocal(photoBox.localToGlobal(photoBox.size.center(Point.origin)));
+    FractionalOffset fractionalCenter = new FractionalOffset(
+      pixelCenter.x / navigatorSize.width,
+      pixelCenter.y / navigatorSize.height
+    );
```

Comment:
Yeah I wasn't really sure how else to do it. We keep getting this problem where we want one thing to depend on the (layout) location of another in the same build.